Applications/TransferLearning/Draw_Classification/jni/main.cpp

```diff
@@ -39,184 +39,201 @@
-      if (strcmp(interpreter->tensor(i)->name, interpreter->GetInputName(j)) ==
-          0)
-        input_idx_list[input_idx_list_len++] = i;
-    }
-    for (int j = 0; j < output_size; j++) {
-      if (strcmp(interpreter->tensor(i)->name, interpreter->GetOutputName(j)) ==
-          0)
-        output_idx_list[output_idx_list_len++] = i;
-    }
-  }
-  for (int i = 0; i < 4; i++) {
-    inputDim[i] = 1;
+  if (input_size > 1 || output_size > 1)
+    throw std::runtime_error("Model is expected with single input and output");
+
+  for (int i = 0; i < MAX_DIM; i++) {
+    tflite_data.inputDimReq[i] = 1;
     outputDim[i] = 1;
   }
 
-  int len = interpreter->tensor(input_idx_list[0])->dims->size;
-  std::reverse_copy(interpreter->tensor(input_idx_list[0])->dims->data,
-                    interpreter->tensor(input_idx_list[0])->dims->data + len,
-                    inputDim);
-  len = interpreter->tensor(output_idx_list[0])->dims->size;
-  std::reverse_copy(interpreter->tensor(output_idx_list[0])->dims->data,
-                    interpreter->tensor(output_idx_list[0])->dims->data + len,
-                    outputDim);
-
-  int output_number_of_pixels = 1;
-  int wanted_channels = inputDim[0];
-  int wanted_height = inputDim[1];
-  int wanted_width = inputDim[2];
-
-  for (int k = 0; k < 4; k++)
-    output_number_of_pixels *= inputDim[k];
-
-  int _input = interpreter->inputs()[0];
+  int input_idx = tflite_data.interpreter->inputs()[0];
+  len = tflite_data.interpreter->tensor(input_idx)->dims->size;
+  std::reverse_copy(tflite_data.interpreter->tensor(input_idx)->dims->data,
+                    tflite_data.interpreter->tensor(input_idx)->dims->data +
+                      len,
+                    tflite_data.inputDimReq);
+
+  int output_idx = tflite_data.interpreter->outputs()[0];
+  len = tflite_data.interpreter->tensor(output_idx)->dims->size;
+  std::reverse_copy(
+    tflite_data.interpreter->tensor(output_idx)->dims->data,
+    tflite_data.interpreter->tensor(output_idx)->dims->data + len, outputDim);
+
+  tflite_data.output_number_of_pixels = 1;
+  for (int k = 0; k < MAX_DIM; k++)
+    tflite_data.output_number_of_pixels *= tflite_data.inputDimReq[k];
+}
 
+/**
+ * @brief     Get Feature vector from tensorflow lite
+ *            This creates interpreter & inference with ssd tflite
+ * @param[in] filename input file path
+ * @param[out] feature_input save output of tflite
+ */
+void getInputFeature(const TFLiteData &tflite_data, const std::string filename,
+                     std::vector<float> &feature_input) {
   uint8_t *in;
-  float *output;
-  in = tflite::label_image::read_bmp(filename, &wanted_width, &wanted_height,
-                                     &wanted_channels);
-  if (interpreter->AllocateTensors() != kTfLiteOk) {
-    std::cout << "Failed to allocate tensors!" << std::endl;
-    exit(0);
+  int inputDim[MAX_DIM] = {1, 1, 1, 1};
+  in = tflite::label_image::read_bmp(filename, inputDim, inputDim + 1,
```

Comment:
Q. our layout is NCHW but it seems the layout is inverted. is this okay?

Reply:
This bmp read data is passed to tensorflow-lite and not to nntrainer.
The output of tflite model is passed to nntrainer which is of shape [1,1,1,128].

Reply:
Got it